src/js/libgeo/GeoOps.js

```diff
@@ -2589,6 +2589,107 @@ geoOps.TransformPolygon.updatePosition = function(el) {
     }));
 };
 
+geoOps.TrComposeTrTr = {};
+geoOps.TrComposeTrTr.kind = "Tr";
+geoOps.TrComposeTrTr.signature = ["Tr", "Tr"];
+geoOps.TrComposeTrTr.updatePosition = function(el) {
+    var a = csgeo.csnames[el.args[0]];
+    var b = csgeo.csnames[el.args[1]];
+    el.matrix = List.normalizeMax(List.productMM(b.matrix, a.matrix));
+    el.dualMatrix = List.normalizeMax(List.productMM(b.dualMatrix, a.dualMatrix));
+};
+
+geoOps._helper.composeMtMt = function(el, m, n) {
```

Comment:
This helps indeed! We should find a way to document the code generated by computer algebra systems. Let's assume github vanishes (unlikely i know, but far from impossible) at some point and we only have the source code left. I don't know if we ever could reconstruct what's the idea behind these lines. Probably the best place would be a comment in the code itself (imo).